### PR TITLE
Mg/windows build

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -79,35 +49,34 @@ def build_extension(self, ext, cmake):
...
 
-with open('README.rst') as f:
+install_requires = ['numpy>=1.14.1',
+                    ]
+
+with open('README.rst', encoding='utf-8') as f:
     long_description = f.read()
 
+if platform.system() == 'Windows':
+    install_requires += ['h5py>=3,<4',   # use h5py's hdf5 install so we don't have to redistrute hdf5
```
Comment (mgeplf) [inline setup.py:70]: redistrute fixed in next patch

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["setuptools", "wheel", "cmake", "ninja"]
+build-backend = "setuptools.build_meta"
```
Comment (matz-e) [inline pyproject.toml:3]: So we're doing this now?

File: setup.py [hunk 3]
```diff
@@ -79,35 +49,34 @@ def build_extension(self, ext, cmake):
...
-                                           "Visual Studio 15 2017 Win64")]
-        else:
-            cmake_args += ['-DCMAKE_BUILD_TYPE={}'.format(cfg),
-                           '-DMorphIO_CXX_WARNINGS=OFF',
-                           ]
-            build_args += ["--", "-j{}".format(max(MIN_CPU_CORES, get_cpu_count())),
-                           ]
+        cmake_args += ['-DCMAKE_BUILD_TYPE={}'.format(cfg),
+                       '-DMorphIO_CXX_WARNINGS=OFF',
+                       '-GNinja',
```
Comment (matz-e) [inline setup.py:54]: Neat (note to self: requires new dependencies in Spack)

**Review by matz-e — APPROVED** [review]
Looks nice, and I appreciate the more compact `setup.py`.

**Comment by matz-e** [comment]
BTW, please change the title of this :)

File: setup.py (hunk 3 above)
Comment (mgeplf) [reply setup.py:54]: yeah, that's why I added you to the review; it should be ok though, right? adding `ninja`?

File: pyproject.toml (hunk 2 above)
Comment (mgeplf) [reply pyproject.toml:3]: This is the way I got things to be 'reliable'; I hate that I don't know the full interactions of this; however, it seems that it doesn't work as well as I expected - I will go back to the drawing board.

File: setup.py (hunk 3 above)
Comment (matz-e) [reply setup.py:54]: `depends_on('ninja', type='build')` should probably be good enough.

**Comment by matz-e** [comment]
Approval does not reset on new pushes?

**Comment by mgeplf** [comment]
> Approval does not reset on new pushes?

apparently not

**Comment by matz-e** [comment]
> > Approval does not reset on new pushes?
> 
> apparently not

Well, I'm still in approval

**Pull request merged**